### PR TITLE
Refactor get_output_from_encoder() to be member of JiantTaskModel

### PR DESCRIPTION
**Pull request by jeswan**
These changes move the if-else statements (for `ModelArchitectures`) in `get_output_from_encoder()` to the newly added `JiantTransformersModel` when using `__call__`.

### CODE REVIEW

File: jiant/proj/main/modeling/primary.py [hunk 1]
```diff
@@ -154,6 +164,19 @@ def get_hidden_size(self):
     def get_hidden_dropout_prob(self):
         return self.config.hidden_dropout_prob
 
+    def __call__(self, input_ids, segment_ids, input_mask, output_hidden_states=True):
```
Comment (zphang) [inline jiant/proj/main/modeling/primary.py:167]: I might have missed this in the previous PR - what's the reasons for modifying the `__class__` and `__call__` of the object? I would prefer a more transparent interface (e.g. `.encode`)

File: jiant/proj/main/modeling/taskmodels.py [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 from jiant.utils.python.datastructures import take_one
-from jiant.shared.model_resolution import ModelArchitectures
+
 from jiant.tasks.core import TaskTypes
 
 
 class JiantTaskModelFactory:
     """This factory is used to create task models bundling the task,
-       encoder, and task head within the task model.
+       jiant_transformers_model, and task head within the task model.
```
Comment (zphang) [inline jiant/proj/main/modeling/taskmodels.py:19]: Likewise, is there a reason for changing from `encoder` to `jiant_transformers_model`? (It looks like it's still using the encoder, not the whole model, so it's a little unclear to me)

Comment (jeswan) [reply jiant/proj/main/modeling/taskmodels.py:19]: The naming change was meant to clarify that it is a `JiantTransformersModel` and not HF's `encoder`. I thought the new name would be more descriptive.

File: jiant/proj/main/modeling/primary.py (hunk 1 above)
Comment (jeswan) [reply jiant/proj/main/modeling/primary.py:167]: It was meant to mimic HF Transformer's forward but I am not attached to any specific interface.

**Pull request merged**